backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v1/bpartner/bpartnercomposite/BPartnerCompositeCacheIndex.java

```diff
@@ -138,4 +148,32 @@ public BPartnerId extractDataItemId(@NonNull final BPartnerComposite dataItem)
 			return result;
 		}
 	}
+
+	@Override
+	public boolean isResetAll(@NonNull final TableRecordReference recordRef)
```

Comment:
not 100% sure, but: if we now have `getRecordRefsRoles`, do we still need to return true if the given recordRef is a `_C_User_Assigned_Role` or `C_User_Role`?

Reply:
yes, we do... as in case of new role being assigned, that new `assigned_role_id` will not be part of the cached key (as it wasn't there before) thus the cache is not refreshed. this behavior is reproducing for every child record where the parent doesn't trigger a refresh when changes occur. @cp-ps will give you more info about this scenario.